### PR TITLE
VMware: Custom Field in vmware_inventory

### LINKED ISSUE #38949: "groupby_custom_field" no longer working when limiting with properties
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

vmware_inventory

##### ANSIBLE VERSION

```
ansible 2.5.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/test/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.14 (default, Mar 14 2018, 13:36:31) [GCC 7.3.1 20180303 (Red Hat 7.3.1-5)]
```

##### CONFIGURATION

```
DEFAULT_FORKS(env: ANSIBLE_FORKS) = 10
DEFAULT_PRIVATE_KEY_FILE(env: ANSIBLE_PRIVATE_KEY_FILE) = /home/test/.config/ansible/.ansible.pem
DEFAULT_REMOTE_USER(env: ANSIBLE_REMOTE_USER) = ansible
DEFAULT_VAULT_PASSWORD_FILE(env: ANSIBLE_VAULT_PASSWORD_FILE) = /home/test/.config/ansible/.vault_pass.txt
RETRY_FILES_ENABLED(env: ANSIBLE_RETRY_FILES_ENABLED) = False
```
##### OS / ENVIRONMENT

N/A
##### SUMMARY

When properties are configured in the ini file,  groupby_custom_field  is no longer available after running the inventory script. 

##### STEPS TO REPRODUCE

Enable groupby_custom_field while limiting data with properties in the ini file. 

```yaml
[vmware]
server=mcm-vc100.example.org
username=svc_em_bind_tower@example.org
validate_certs = False
cache_max_age = 1800
cache_path = ~/.cache/ansible/ent_services
max_object_level=2
host_filters={{ config.guestid == 'rhel7_64Guest' }},{{ 'ace-' in config.name or 'mce-' in config.name }}
groupby_custom_field = True

[properties]
prop01=name
#prop02=config.cpuHotAddEnabled
#prop03=config.cpuHotRemoveEnabled
prop04=config.instanceUuid
#prop05=config.hardware.numCPU
prop06=config.template
prop07=config.name
prop08=guest.hostName
prop09=guest.ipAddress
prop10=guest.guestId
#prop11=guest.guestState
#prop12=runtime.maxMemoryUsage
prop13=config.guestId
prop14=vm.availableField
prop15=vm.customValue
prop16=config.uuid
prop17=vm.value
```

##### EXPECTED RESULTS

```
Output without limited properties:
"rhel7_64Guest": {
    "hosts": [
      "mce-server101_12345678-a12b-cdef-1234-12345678ac11"
    ]
  }, 
  "all": {
    "hosts": [
      "mce-server101_12345678-a12b-cdef-1234-12345678ac11"
    ]
  }, 
  "vmware_tag_ANSIBLE_GROUP_splunk_servers": {
    "hosts": [
      "mce-server101_12345678-a12b-cdef-1234-12345678ac11"
    ]
  },
```
##### ACTUAL RESULTS

```
Output with limited properties:
"rhel7_64Guest": {
    "hosts": [
      "mce-server101_12345678-a12b-cdef-1234-12345678ac11"
    ]
  }, 
  "all": {
    "hosts": [
      "mce-server101_12345678-a12b-cdef-1234-12345678ac11"
    ]
  },
```

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Fixes: #38949

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
contrib/inventory/vmware_inventory.py

##### ANSIBLE VERSION

```
2.6-devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
@pdellaert @dericcrago Could you please review this ?

File: contrib/inventory/vmware_inventory.py [hunk 1]
```diff
@@ -494,16 +494,13 @@ def instances_to_inventory(self, instances):
             for k, v in inventory['_meta']['hostvars'].items():
                 if 'customvalue' in v:
                     for tv in v['customvalue']:
-                        if not isinstance(tv['value'], string_types):
```
Comment (pdellaert) [inline contrib/inventory/vmware_inventory.py:497]: Why is this being removed?

Comment (Akasurde) [reply contrib/inventory/vmware_inventory.py:497]: When groupby_custom_fields is used, customValues can be of type `vim.CustomFieldsManager.Value`. Due to this block of code, script does not go beyond this point. I am not sure why this was added.

**Review by pdellaert — APPROVED** [review]

**Comment by Akasurde** [comment]
@pdellaert Thanks for the review.

**Pull request merged**